aziotctl/aziotctl-common/src/system/restart.rs

```diff
@@ -35,3 +40,33 @@ fn start(name: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(feature = "snapctl")]
+pub fn restart(services: &[&ServiceDefinition]) -> Result<()> {
+    let snap_instance_name = match std::env::var("SNAP_INSTANCE_NAME") {
+        Ok(snap_instance_name) => snap_instance_name,
+        Err(_) => {
+            std::env::var("SNAP_NAME").expect("snapctl must be used within the context of a snap")
+        }
+    };
+
+    print!("Restarting {} services...", snap_instance_name);
```

Comment:
Sure. I think the only thing is that since the systemctl version just calls the stop & start functions, whereas this makes a single call to snapctl restart, it still won't have parity because the non-snapctl version will have stop & start messages whereas this will have no messages. Let me know if this is ok/what you think the best way to handle it would be

Reply:
Does snapctl have an equivalent start/stop that could be used?

Reply:
It does, and I used that to implement the stop command. I can change this to just call stop then start